scipy/stats/_hypotests.py

```diff
@@ -130,3 +131,133 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
     p = chi2.sf(w, r)
 
     return Epps_Singleton_2sampResult(w, p)
+
+
+def poisson_etest(k1, k2, n1, n2, diff=0, alternative='two-sided'):
+    """
+    Calculate E-test for the mean difference of two samples
+
+    The test requires two samples coming from poisson distribution. This is a
+    right-sided and two-sided test.
+
+    Parameters
+    ----------
+    k1, k2 : int or float
+        Count from the first and second samples respectively
+    n1, n2 : int or float
+        Sample size of each sample
+    diff : int of float, optional
+        The difference of mean between two samples under null hypothesis
+    alternative : {'two-sided', 'greater'}, optional
+        Whether to get p-value for two-sided hypothesis or right sided hypothesis.
+        Right-sided test that mean from sample one is greater than sample two
+
+    Returns
+    -------
+    pvalue : float
+        The associated p-value based on estimated p-value of the standardized
+        difference.
+
+    Notes
+    -----
+    The Poisson distribution is commonly used to model many processes such as
+    transactions per user. The usual test to compare difference between two
+    means of Poisson samples is C-test, based on conditional distribution.
+    Meanwhile the E-test is an unconditional test
+
+    Based the author results [1]_, E-test is more powerful than C-test. The E-test
+    is almost exact because the test exceed the nominal value only by negligible
+    amount. Compared to C-test which produce smaller size than nominal value.
+
+    References
+    ----------
+    .. [1]  https://userweb.ucs.louisiana.edu/~kxk4695/JSPI-04.pdf
+
+    Examples
+    --------
+    >>> from scipy import stats
+
+    Taken from Przyborowski and Wilenski (1940). Suppose that a purchaser wishes to
+    test the number of dodder seeds (a weed) in a sack of clover seeds that he bought
+    from a seed manufacturing company. A 100 g sample is drawn from a sack of clover
+    seeds prior to being shipped to the purchaser. The sample is analyzed and found to
+    contain no dodder seeds; that is, k1 = 0. Upon arrival, the purchaser also draws
+    a 100 g sample from the sack. This time, three dodder seeds are found in the sample;
+    that is, k2 = 3. The purchaser wishes to determine if the difference between the
+    samples could not be due to chance.
+
+    >>> stats.poisson_etest(0, 3, 100, 100)
+    0.08837900929018155
```

Comment:
please add a conclusion of the test. i think it would be informative to add the results of `ttest_ind_from_stats`

Reply:
added the conclusion and t-test result, + extend into large vs small sample size results

    >>> stats.poisson_etest(0, 3, 100, 100)
    0.08837900929018155
    >>> stats.ttest_ind_from_stats(mean1=0, std1=0, nobs1=100, equal_var=False,
    ...                            mean2=3.0/100, std2=np.sqrt(.03), nobs2=100)
    Ttest_indResult(statistic=-1.7320508075688772, pvalue=0.0863790757063214)

    The result above show evidence that the difference between two samples is significant
    at the level of significance 0.1. Both t-test and e-test show similar p-value, but
    will be different if the sample size is small

    >>> stats.poisson_etest(0, 3, 10, 10)
    0.0883790094548346
    >>> stats.ttest_ind_from_stats(mean1=0, std1=0, nobs1=10, equal_var=False,
    ...                            mean2=3.0/10, std2=np.sqrt(.3), nobs2=10)
    Ttest_indResult(statistic=-1.7320508075688774, pvalue=0.11730680301423814)

    The t-test need n=100 to arrive at the same p-value of e-test whereas the e-test
    produce similar p-value for n=10 and n=100